### ISSUE TITLE
Update index.md

### ISSUE THREAD

**Issue by kirillbobyrev** [llvm/clangd-www#30]
The index page looks empty and is not really useful for understanding what Clangd is/how it helps you. Completion is cool but does not advertise Clangd much from the first sight (compiler and Clang-Tidy diags + rename might do that). We can either:

- Update it: add more info and a couple of interactive demos, link "Features" (this is what most users would be interested in anyway, right?)
- Point index to the features directly and call it differently/make it "index-aware" (assume this is the first page users land on).

**Comment by kirillbobyrev** [comment]
@sam-mccall @kadircet WDYT?

**Comment by kadircet** [comment]
yes, that definitely SGTM!

**Comment by kirillbobyrev** [comment]
> yes, that definitely SGTM!

which one do you prefer, 1 or 2?

**Comment by kadircet** [comment]
ah sorry they both felt like the same thing at first glance.

I suppose having a page which describes features at a glance, with links to pages with more details whenever possible would be great. (i think that's what you mean by first option)

If we can generate such a page automatically (i suppose that's what you mean by the second one?), that would be even better. i am just unaware if there are any solutions offered by github-pages for such a thing though, and i am not sure if it would be worth chasing something more complex.

So I'd definitely opt for first one, unless there's an easy solution for the second.

**Comment by wusatosi** [comment]
It may actually make sense for the index page be the same as the features page.

**Comment by kirillbobyrev** [comment]
> It may actually make sense for the index page be the same as the features page.

I don't think it should be _exactly_ the same page as the features simply because feature overview doesn't explain what clangd is at all and goes directly into details. But overall, yes, I think it should be possible to just merge two of them.

**Comment by wusatosi** [comment]
> > It may actually make sense for the index page be the same as the features page.
> 
> I don't think it should be _exactly_ the same page as the features simply because feature overview doesn't explain what clangd is at all and goes directly into details. But overall, yes, I think it should be possible to just merge two of them.

Yeah, merging them sounds like a good idea, will `{% include features.md %}` on `index.md` work? 
I can submit a pr if this is good enough.

**Comment by wusatosi** [comment]
> > It may actually make sense for the index page be the same as the features page.
> 
> I don't think it should be _exactly_ the same page as the features simply because feature overview doesn't explain what clangd is at all and goes directly into details. But overall, yes, I think it should be possible to just merge two of them.

Hi, @kirillbobyrev , I did a proof of concept at #83 . Have a look and tell me what do you think!
